File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    person_dict = {}
    
    for i in participant:
        if i in person_dict:
            person_dict[i] = person_dict[i] + 1
        else:
            person_dict[i] = 1
    
    for i in completion:
        if i in person_dict:
            person_dict[i] = person_dict[i] - 1
    
    for i in person_dict:
        if person_dict[i] > 0:
            return i
        

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) { 
    const people = {}
    
    participant.forEach(p => {
        if (people[p]) {
            people[p] += 1
        } else {
            people[p] = 1
        }
    })
    
    completion.forEach(p => {
        if (people[p]) {
            people[p] -= 1
        }
    })
    
    return Object.entries(people).filter(([key, value]) => value === 1)[0][0]
}